tortoise/fields.py

```diff
@@ -34,13 +43,9 @@ def __init__(
         self.unique = unique
 
     def to_db_value(self, value):
-        if value is None or isinstance(value, self.type):
-            return value
-        return self.type(value)
+        return value
 
     def to_python_value(self, value):
-        if value is None or isinstance(value, self.type):
-            return value
```

Comment:
I think this should stay, I came up with the bug that we currently having on `values` and `values_list` - they don't serialise results with `to_python_value`. I came up with such test to demonstrate it:
```
    async def test_json_values(self):
        data = {'some': ['text', 3]}
        obj0 = await testmodels.JSONFields.create(data=data)
        values = await testmodels.JSONFields.filter(
            id=obj0.id,
        ).values('data')
        self.assertDictEqual(values[0]['data'], data)    # fails because values[0]['data'] is string
```

Could you please return this assertion of `is None` and fix `values` and `values_list` executing to serialise results

Reply:
The `to_python_value` methods are only called from `tortoise/models.py` L333
and there it is after a guard like so:
```py
 if not isinstance(value, field_object.type) and value is not None:
    setattr(self, key, field_object.to_python_value(value))
```
So that logic is duplicate.
Adding that code I took away back doesn't resolve it.

What I did wrong there is defining the type of the JSONField, as only `(dict, list)` whereas JSON supports more base types. The problem comes in with it being a `str` that is how it knows it should serialise/deserialise. I feel to solve this properly, instead of a "to_db_value" and a "to_python_value" we should change it to be a "to_db_value" and "from_db_value" (in principal).
But the current system is better ito lazy evaluation...

Another short-term possibility is to state that JSONField only works for dicts/lists as the root element?

Reply:
Oh, I missed your point. It is a good one, and I should add _values and _values_list tests for all fields.
We might have to consider using to_python_value for that, as date/datetime would also cause issues for SQLite...

What I said about it assuming a 'str' is the serialised json object is still valid for that use case.